src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractWorkspacePullDiagnosticsHandler.cs

```diff
@@ -130,6 +139,24 @@ protected override async Task WaitForChangesAsync(RequestContext context, Cancel
         return result.ToImmutable();
     }
 
+    /// <summary>
+    /// There are three potential sources for reporting workspace diagnostics:
+    ///
+    ///  1. Full solution analysis: If the user has enabled Full solution analysis, we always run analysis on the latest
+    ///                             project snapshot and return up-to-date diagnostics computed from this analysis.
+    ///
+    ///  2. Code analysis service: Otherwise, if full solution analysis is disabled, and if we have diagnostics from an explicitly
+    ///                            triggered code analysis execution on either the current or a prior project snapshot, we return
+    ///                            diagnostics from this execution. These diagnostics may be stale with respect to the current
+    ///                            project snapshot, but they match user's intent of not enabling continuous background analysis
+    ///                            for always having up-to-date workspace diagnostics, but instead computing them explicitly on
+    ///                            specific project snapshots by manually running the "Run Code Analysis" command on a project or solution.
+    ///
+    ///  3. EnC analysis: Emit and debugger diagnostics associated with a closed document or not associated with any document.
+    ///
+    /// If full solution analysis is disabled AND code analysis was never executed for the given project,
+    /// we have no workspace diagnostics to report and bail out.
+    /// </summary>
     public static async ValueTask<ImmutableArray<IDiagnosticSource>> GetDiagnosticSourcesAsync(
```

Comment:
are all the changes from this point to the end of the file just cleanup?  or does any logic change?  i ask because it looksl ike cleanup only, but i don't want to miss anythign important.  if logic changed, can you call it out?  thanks!

Reply:
Logic didn't change. Just cleanup.

Reply:
Great.  Thanks for the assurance.